core/src/main/java/org/kiwix/kiwixmobile/core/di/modules/NetworkModule.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package org.kiwix.kiwixmobile.core.di.modules
+
+import android.content.Context
+import android.net.ConnectivityManager
+import dagger.Module
+import dagger.Provides
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import okhttp3.logging.HttpLoggingInterceptor.Level.BASIC
+import okhttp3.logging.HttpLoggingInterceptor.Level.NONE
+import org.kiwix.kiwixmobile.core.BuildConfig
+import org.kiwix.kiwixmobile.core.data.remote.KiwixService
+import org.kiwix.kiwixmobile.core.data.remote.KiwixService.ServiceCreator
+import org.kiwix.kiwixmobile.core.data.remote.UserAgentInterceptor
+import java.util.concurrent.TimeUnit.SECONDS
+import javax.inject.Singleton
+
+@Module
+open class NetworkModule {
+  @Suppress("MagicNumber")
```

Comment:
here magic number is very much applicable and it would be handy to have the timeouts in constants we could reference and easily change